### ISSUE TITLE
'Custom' network architecture

### ISSUE THREAD

**Issue by imankgoyal** [reinforceio/tensorforce#488]
Is there a way to plug in a 'custom' network architecture that maps the states into actions. I am aware that the library supports network architectures which are specified as a list of layers. But suppose, the architecture has multiple channels and looping (like Dynamic Memory Network). Is it possible to plug in such architecture? Can I, for example, create a neural network architecture in Tensorflow, specify the tensors representing states and actions, and plug it in?

**Comment by AlexKuhnle** [comment]
Yes, that's possible. On the one hand, for _somewhat_ more complex architectures with multiple inputs, the network representation also accepts lists of layer-lists, i.e. composed layered sub-networks, see e.g. [here](https://github.com/reinforceio/tensorforce/blob/master/tensorforce/tests/base_agent_test.py#L134). On the other hand, for more fancy custom implementations you have to implement your own [Network](https://github.com/reinforceio/tensorforce/blob/master/tensorforce/core/networks/network.py#L30) or, maybe more helpful, [LayerBasedNetwork](https://github.com/reinforceio/tensorforce/blob/master/tensorforce/core/networks/network.py#L150). You can then pass your custom network class `CustomNetwork` to the agent argument `network=CustomNetwork` instead of a layer list. Note, however, that the action distributions plus required linear layers are added in TensorForce internally, so the output of your network needs only be an arbitrarily-sized tensor of shape `[batch, size]` (the bigger, the more information for the distribution) and you don't need to take care of distribution-things like `softmax` etc.

**Comment by batu** [comment]
@imankgoyal Were you able to go about doing this? If so, can you give me a high level overview to replicate it?

Thank you!
